plugins/woocommerce/tests/e2e-pw/tests/basic/page-loads.spec.js

```diff
@@ -216,35 +223,59 @@ for ( const currentPage of wcPages ) {
 			} )
 			.then( ( r ) => {
 				orderId = r.data.id;
+			} )
+			.catch( ( e ) => {
+				console.error(
+					`Failed to create order ${
+						e.data ? JSON.stringify( e.data ) : ''
+					}`
+				);
+				throw e;
 			} );
-
-		// create customer
-		await restApi
-			.post( `${ WC_API_PATH }/customers`, customer )
-			.then( ( r ) => ( customer.id = r.data.id ) );
 	} );
 
 	test.afterAll( async ( { restApi } ) => {
-		await restApi.delete( `${ WC_API_PATH }/orders/${ orderId }`, {
-			force: true,
-		} );
-		await restApi.delete( `${ WC_API_PATH }/products/${ product.id }`, {
-			force: true,
-		} );
-		await restApi.delete( `${ WC_API_PATH }/customers/${ customer.id }`, {
-			force: true,
-		} );
+		await restApi
+			.delete( `${ WC_API_PATH }/orders/${ orderId }`, {
+				force: true,
+			} )
+			.catch( ( e ) => {
+				console.error(
+					`Failed to delete order ${
+						e.data ? JSON.stringify( e.data ) : ''
+					}`
+				);
+				throw e;
+			} );
+		await restApi
+			.delete( `${ WC_API_PATH }/products/${ product.id }`, {
+				force: true,
+			} )
+			.catch( ( e ) => {
+				console.error(
+					`Failed to delete product ${
+						e.data ? JSON.stringify( e.data ) : ''
+					}`
+				);
+				throw e;
+			} );
 	} );
 
 	for ( let i = 0; i < currentPage.subpages.length; i++ ) {
 		test( `can load ${ currentPage.name } > ${ currentPage.subpages[ i ].name } page`, async ( {
 			page,
 		} ) => {
 			await page.goto( currentPage.url );
+
+			// needs a Regexp on link name to match exact text and also match the possible counter
+			// E.g. should match "Orders 3" or "Orders", but should not match "Quick Orders"
 			await page
-				.locator(
-					`li.wp-menu-open > ul.wp-submenu > li a:has-text("${ currentPage.subpages[ i ].name }")`
-				)
+				.locator( 'li.wp-menu-open > ul.wp-submenu' )
+				.getByRole( 'link', {
+					name: new RegExp(
+						`^${ currentPage.subpages[ i ].name }( \\d+)?$`
```

Comment:
This might apply to every item in the WP-Admin menu, which means we might want to refactor existing tests to follow this approach or keep it in mind for new tests. Do you think it would be helpful to extract this code to a utils function we can reuse?

Something like:
```
function getPageName() {
    return new RegExp( `^${ currentPage.subpages[ i ].name }( \\d+)?$` );
}
```

Reply:
I don't know of any other test using this, so we're good. Usually navigation is done with direct link to the page.

Reply:
Ah cool!